LinuxCommunicator/CallbackController.cs

```diff
@@ -84,5 +82,35 @@ public int[] GetInstanceIds(string nodeName, [FromBody] string[] instanceNames)
 
             return null;
         }
+
+        [HttpGet]
+        [Route("api/hostfile")]
+        public HttpResponseMessage GetHosts()
+        {
+            Guid curUpdateId = LinuxCommunicator.Instance.HostsManager.UpdateId;
+            string updateid = string.Empty;
+            IEnumerable<string> updateids;
+            if (Request.Headers.TryGetValues("UpdateId", out updateids))
+            {
+                foreach (var id in updateids)
+                {
+                    updateid = id;
+                }
+            }
+
+            HttpResponseMessage response = null;
+            Guid guid;
+            if (Guid.TryParse(updateid, out guid) && guid == curUpdateId)
```

Comment:
Suggest remove the updateid definition, and write directly: Guid.TryParse(updateIds.FirstOrDefault(), out id) && ...